triggermail_templates.py

```diff
@@ -128,6 +134,53 @@ def dissect_filename(self, template_filename):
         self.partner = self.settings.get("partner", self.partner) or self.partner
         self.partner = self.partner.replace("_templates", "")
 
+    def generate_pertinent_file_names_and_map(self, template_filename):
+        template_filename = template_filename.replace(self.path + '/', '')
+        image_regex = "/img/" + self.partner + "/" + "(.*?)['\"]"
+        file_map = dict()
+        all_file_names_required = [template_filename]
+        all_images_required = []
+        # go through all files this template requires
+        # we get all the filenames, their contents, and images
+        for file_name in all_file_names_required:
+            # some templates include html files that don't exist. Just because they don't exist, doesn't
+            # mean it should automatically fail the command, they are most likely in jinja "if" statements
+            try:
+                contents = read_file(os.path.join(self.path, file_name))
+            except:
+                continue
+            file_map[file_name] = contents
+            # this regex finds all the other files the html file depends on. Because we're thenadding to
+            # the loop we're cycling through, this can be recursive
+            files_required_by_file = [file_name for _,file_name in \
+                                        re.findall("{%\s*(include|extends|import)\s*['\"](.*?)['\"]", contents)]
+            all_file_names_required += [x for x in files_required_by_file if x not in all_file_names_required]
+            images_required_by_file = re.findall(image_regex, contents)
+            all_images_required += [img for img in re.findall(image_regex, contents) if img not in all_images_required]
+
+        for img in all_images_required:
+            img_path = os.path.abspath(os.path.join(self.image_path, img))
+            contents = encode_image(img_path)
+            file_map[img] = contents
+
+        all_file_names_required += all_images_required
+
+        # lastly, we need to go through the other campaign files that accompany the html
+        template_base_name = template_filename.split('.')[0]
```

Comment:
You might want to prefer `os.path.splitext` because it will "correctly" handle files named things like `.profile`. In this case it might not matter, so maybe never mind. https://docs.python.org/2/library/os.path.html#os.path.splitext